### PR TITLE
[Java] ID length fix

### PR DESCRIPTION
**Pull request by kfstorm**
## Why are these changes needed?

https://github.com/ray-project/ray/pull/6425 doesn't include changes for Java.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: java/api/src/main/java/org/ray/api/id/JobId.java [hunk 1]
```diff
@@ -38,11 +35,11 @@ public static JobId fromByteBuffer(ByteBuffer bb) {
     return new JobId(byteBuffer2Bytes(bb));
   }
 
-  public static JobId fromInt(int value) {
+  public static JobId fromShort(short value) {
```
Comment (raulchen) [inline java/api/src/main/java/org/ray/api/id/JobId.java]: As discussed offline, 2 bytes is too short for job id. @edoakes will increase this length.
So we can still keep the value as an integer.

**Comment by MissiontoMars** [comment]
Can we use something like [SWIG](http://www.swig.org/) to wrap these c++ id objects to simple Java/Python objects? Changes in the c++ code will not cause Java/Python change.

**Comment by kfstorm** [comment]
@MissiontoMars The current implementation of single process mode in Java worker is pure Java, no JNI involved. Does the tool you mentioned works like JNI or as a build tool to generate Java/python code as a part of build process? Do we still need to run a script manually if `id.h` changed?

File: java/api/src/main/java/org/ray/api/id/JobId.java (hunk 1 above)
Comment (kfstorm) [reply java/api/src/main/java/org/ray/api/id/JobId.java]: Sure, we can hold this PR.

**Comment by MissiontoMars** [comment]
> @MissiontoMars The current implementation of single process mode in Java worker is pure Java, no JNI involved. Does the tool you mentioned works like JNI or as a build tool to generate Java/python code as a part of build process? Do we still need to run a script manually if `id.h` changed?

Base on Jni. The jni shared library libcore_worker_library_java.so is copied to /tmp/ automatically， so maybe there is no conflict between the single process mode and jni? Another problem is when our users run single process mode they also have to compile jni, but It's not a big problem under bazel.

**Review by edoakes — APPROVED** [review]

**Review by raulchen — APPROVED** [review]

**Pull request merged**